src/pptr-utils/get-links.ts

```diff
@@ -22,10 +27,15 @@ export const getLinks = async (page): Promise<LinkObject[]> => {
 
 // https://dev.to/vuevixens/removing-duplicates-in-an-array-of-objects-in-js-with-sets-3fep
 export const dedupLinks = (links_with_duplicates: LinkObject[]) => {
-    const links = Array.from(new Set(links_with_duplicates.filter(f => f && f.hasOwnProperty('href')).map(link => link.href))).map(href => {
-        return links_with_duplicates.find(link => link.href === href);
-    });
-
+    const sanitizedLinks = links_with_duplicates.filter(f => f && hasOwnProperty(f, 'href'));
+    const dedupedLinkArray = Array.from(new Set(sanitizedLinks));
+    // I don't think the bellow modification actually does anything,
+    // but I'm gonna write tests for this function before pulling the plug
+    const links = dedupedLinkArray
+                    .map((link:LinkObject) => link.href)
+                    .map(href => {
+                        return links_with_duplicates.find(link => link.href === href);
+                    });
```

Comment:
This is much more readable, thank you. My first impression of this is the same: `links` gets set to `dedupedLinkArray` filtered for links that also exist in the original input array. Seems like an unnecessary check since `dedupedLinkArray` is already derived from `links_with_duplicates`.